src/niscope/system_tests/test_system_niscope.py

```diff
@@ -1,527 +1,588 @@
-    assert last_cal.month == 12
-    assert last_cal.day == 21
-    assert last_cal.year == 1999
-    assert last_cal.hour == 0
-    assert last_cal.minute == 0
-
-
-def test_get_ext_cal_last_date_time(single_instrument_session):
-    last_cal = single_instrument_session.get_ext_cal_last_date_and_time()
-    assert last_cal.month == 12
-    assert last_cal.day == 21
-    assert last_cal.year == 1999
-    assert last_cal.hour == 0
-    assert last_cal.minute == 0
-
-
-def test_get_self_cal_last_temperature(single_instrument_session):
-    last_cal_temp = single_instrument_session.get_self_cal_last_temp()
-    assert last_cal_temp == 25
-
-
-def test_get_ext_cal_last_temperature(single_instrument_session):
-    last_cal_temp = single_instrument_session.get_ext_cal_last_temp()
-    assert last_cal_temp == 25
-
-
-def test_self_test(multi_instrument_session):
-    # We should not get an assert if self_test passes
-    multi_instrument_session.self_test()
-
-
-def test_reset(multi_instrument_session):
-    deault_fetch_relative_to = multi_instrument_session._fetch_relative_to
-    assert deault_fetch_relative_to == niscope.FetchRelativeTo.PRETRIGGER
-    multi_instrument_session._fetch_relative_to = niscope.FetchRelativeTo.READ_POINTER
-    non_default_acqusition_type = multi_instrument_session._fetch_relative_to
-    assert non_default_acqusition_type == niscope.FetchRelativeTo.READ_POINTER
-    multi_instrument_session.reset()
-    assert multi_instrument_session._fetch_relative_to == niscope.FetchRelativeTo.PRETRIGGER
-
-
-def test_reset_device(multi_instrument_session):
-    deault_meas_time_histogram_high_time = multi_instrument_session.meas_time_histogram_high_time
-    assert deault_meas_time_histogram_high_time == hightime.timedelta(microseconds=500)
-    multi_instrument_session.meas_time_histogram_high_time = hightime.timedelta(microseconds=1000)
-    non_default_meas_time_histogram_high_time = multi_instrument_session.meas_time_histogram_high_time
-    assert non_default_meas_time_histogram_high_time == hightime.timedelta(microseconds=1000)
-    multi_instrument_session.reset_device()
-    assert multi_instrument_session.meas_time_histogram_high_time == hightime.timedelta(microseconds=500)
-
-
-def test_reset_with_defaults(multi_instrument_session):
-    deault_meas_time_histogram_high_time = multi_instrument_session.meas_time_histogram_high_time
-    assert deault_meas_time_histogram_high_time == hightime.timedelta(microseconds=500)
-    multi_instrument_session.meas_time_histogram_high_time = hightime.timedelta(microseconds=1000)
-    non_default_meas_time_histogram_high_time = multi_instrument_session.meas_time_histogram_high_time
-    assert non_default_meas_time_histogram_high_time == hightime.timedelta(microseconds=1000)
-    multi_instrument_session.reset_device()
-    assert multi_instrument_session.meas_time_histogram_high_time == hightime.timedelta(microseconds=500)
-
-
-def test_error_message():
```

Comment:
Two different versions of `test_error_message` due to the different setup. One in `TestLibrary` and one in `TestGrpc`

Reply:
Acknowledged. The test creates its own session, so for the moment TestGrpc has its own copy of the method in order to pass grpc_options as session creation. This makes sense.